kubevirt-vmware/pkg/controller/ovirtprovider/client.go

```diff
@@ -0,0 +1,192 @@
+package ovirtprovider
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	ovirtsdk "github.com/ovirt/go-ovirt"
+	kubevirtv1alpha1 "github.com/ovirt/v2v-conversion-host/kubevirt-vmware/pkg/apis/v2v/v1alpha1"
+)
+
+// Client struct holding implementation details required to interact with oVirt engine
+type Client struct {
+	conn    *ovirtsdk.Connection
+	ctx     context.Context
+	Cluster string
+}
+
+// NewClient creates new client struct based on connection details provided
+func NewClient(ctx context.Context, url string, username string, password string, cluster string) (*Client, error) {
+	conn, err := ovirtsdk.NewConnectionBuilder().
+		URL(url).
+		Username(username).
+		Password(password).
+		// TODO: check how can we provide CA in the UI
+		Insecure(true).
```

Comment:
https://github.com/oVirt/v2v-conversion-host/pull/75#discussion_r390133793

\+ 1 ca  can be optional in the secret

Reply:
OK, will add.